### DISCUSSION TITLE
Can not build a navier stokes app

### DISCUSSION THREAD

**Discussion by majumdararn** [idaholab/moose#18108] [Q&A Modules: Navier-Stokes] [answered]
I want to build an app with the Navier_Stokes module on. When I make this app, it shows following error.

/home/amajumda/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/amajumda/moose/modules/module_loader/lib/libmodule_loader_with_fp_ray_hc_rdg_ns-opt.so: undefined reference to `NavierStokesApp::registerExecFlags(Factory&)'
/home/amajumda/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/amajumda/moose/modules/module_loader/lib/libmodule_loader_with_fp_ray_hc_rdg_ns-opt.so: undefined reference to `NavierStokesApp::registerObjects(Factory&)'
/home/amajumda/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/amajumda/moose/modules/module_loader/lib/libmodule_loader_with_fp_ray_hc_rdg_ns-opt.so: undefined reference to `NavierStokesApp::associateSyntax(Syntax&, ActionFactory&)'
collect2: error: ld returned 1 exit status
make: *** [/home/amajumda/moose/framework/app.mk:407: /home/amajumda/navier_stokes/navier_stokes-opt] Error 1

Could anyone help?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Let’s first try to update moose and libmesh. 
for moose pull from the repository, for lib mesh run conda update —all

then `make clobberall` to clean up and `make -j<the number of cores>` again

Guillaume

**Reply by majumdararn** [reply]
@GiudGiud Thank you very much for your suggestion
It worked.
I ran git pull inside moose directory and then did conda update --all
finally did make clobberall and then make -j<Number of Cores>
